### PR TITLE
Use typescript in typescript plugin

### PR DESCRIPTION
**Pull request by NotWoods**
## Rollup Plugin Name: `typescript`

This PR contains:

- [ ] bugfix
- [ ] feature
- [x] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [x] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

List any relevant issue numbers:

### Description

Refactoring the TypeScript plugin to use TypeScript itself. This is to prepare for larger refactors to add new features down the line.

This also fixes the types for options.tslib to reflect what is actually supported.

### CODE REVIEW

File: packages/typescript/rollup.config.js [hunk 1]
```diff
@@ -1,21 +1,15 @@
-import buble from '@rollup/plugin-buble';
+import typescript from '@rollup/plugin-typescript';
```
Comment (pnevares) [inline packages/typescript/rollup.config.js]: Shouldn't this be a relative import from local code?

File: packages/typescript/package.json [hunk 2]
```diff
@@ -49,17 +49,25 @@
   },
   "dependencies": {
     "@rollup/pluginutils": "^3.0.0",
-    "resolve": "^1.12.2"
+    "resolve": "^1.13.1"
   },
   "devDependencies": {
     "@rollup/plugin-buble": "^0.20.0",
+    "@rollup/plugin-typescript": "^2.0.1",
```
Comment (pnevares) [inline packages/typescript/package.json]: This definitely feels unusual 🤔

File: packages/typescript/rollup.config.js (hunk 1 above)
Comment (NotWoods) [reply packages/typescript/rollup.config.js]: Making it a relative import depends on the code already being compiled in order to compile it. We kind of follow this pattern already as we don't depend on relative packages in the monorepo, instead depending on the npm version of pluginutils, babel, etc.

Comment (pnevares) [reply packages/typescript/rollup.config.js]: Thanks, I must have never noticed! 👍

**Comment by shellscape** [comment]
Looks like an eslint comment is needed for a fixture to pass linting

**Pull request merged**